docs/source/nlp/punctuation_and_capitalization.rst

```diff
@@ -158,55 +162,702 @@ for the training and evaluation in the :code:`conf/punctuation_capitalization_co
+   * - **optim**
+     - :ref:`optim config<optim-config-label>`
+     - :ref:`optim config<optim-config-label>`
+     - A configuration of optimizer, learning rate scheduler, and L2 regularization.
+
+.. _class-labels-config-label:
+
+Class labels config
+^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Location of class labels config in parent configs
+   :widths: 5 5
+   :header-rows: 1
+
+   * - **Parent config**
+     - **Key in parent config**
+   * - :ref:`Run config<run-config-label>`
+     - ``model.class_labels``
+   * - :ref:`Model config<model-config-label>`
+     - ``class_labels``
+
+.. list-table:: Class labels config
+   :widths: 5 5 5 35
+   :header-rows: 1
+
+   * - **Parameter**
+     - **Data type**
+     - **Default value**
+     - **Description**
+   * - **punct_labels_file**
+     - string
+     - ???
+     - A name of a punctuation labels file. It is a mandatory parameter that cannot be omitted in the config. This name
+       is used as a name of label ids file in ``.nemo`` checkpoint. It also can be used for passing label vocabulary to
+       the model. If ``punct_labels_file`` is used as a vocabulary file, then you should provide parameter
+       ``label_vocab_dir`` in :ref:`common dataset parameters<common-dataset-parameters-config-label>`
+       (``model.common_dataset_parameters.label_vocab_dir`` in :ref:`run config<run-config-label>`). Each line of
+       ``punct_labels_file`` file contains 1 label. The first line has to contain neutral label which has to be
+       equal to a ``pad_label`` parameter in :ref:`common dataset parameters<common-dataset-parameters-config-label>`.
+   * - **capit_labels_file**
+     - string
+     - ???
+     - Same as ``punct_labels_file`` for capitalization labels.
+
+.. _common-dataset-parameters-config-label:
+
+Common dataset parameters config
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Location of common dataset parameters config in parent config
+   :widths: 5 5
+   :header-rows: 1
+
+   * - **Parent config**
+     - **Key in parent config**
+   * - :ref:`Run config<run-config-label>`
+     - ``model.common_dataset_config``
+   * - :ref:`Model config<model-config-label>`
+     - ``common_dataset_config``
+
+A common dataset parameters config which includes label and loss mask information.
+If you omit parameters ``punct_label_ids``, ``capit_label_ids``, ``label_vocab_dir``, then labels will be inferred
+from a training dataset or loaded from a checkpoint.
+
+Parameters ``ignore_extra_tokens`` and ``ignore_start_end`` are responsible for forming loss mask. A loss mask
+defines on which tokens loss is computed.
+
+.. list-table:: Common dataset parameters config
+   :widths: 5 5 5 35
+   :header-rows: 1
+
+   * - **Parameter**
+     - **Data type**
+     - **Default value**
+     - **Description**
+   * - **pad_label**
+     - string
+     - ???
+     - A mandatory parameter which should contain label used for punctuation and capitalization label padding. It
```

Comment:
there is a default value as well - I would avoid the word `mandatory` for config args with default values that users wouldn't want to change.